### PR TITLE
Add Dark mode toggler

### LINKED ISSUE #6: Feature : Add dark mode light node toggler
Add a dark mode, light mode theme toggler for the editor

### PR DESCRIPTION
**Pull request by arp99**
Closes #6 
- [x] Add a theme toggler to switch editor styles from light mode to dark mode and vice versa
- [x] Theme remain intact on reloads

capture-editor

### CODE REVIEW

**Comment by arp99** [comment]
@NegiAkash890 Please review the PR

**Comment by NegiAkash890** [comment]
Hi @arp99 I see you have done some good work but since we have two PRs using redux. I am actually thinking whether we can combine the redux together somehow. Tagging PR #7 . Let me know your view on same?

**Comment by arp99** [comment]
> Hi @arp99 I see you have done some good work but since we have two PRs using redux. I am actually thinking whether we can combine the redux together somehow. Tagging PR #7 . Let me know your view on same?

I have seen the PR using REDUX, but I do not think so you need to use REDUX for that small change, where you only keeping some boilerplate code, that's unnecessary, For smaller apps like this context does all the work and also updating everything on the global; store is not always necessary as it might make the app less performant

**Comment by arp99** [comment]
And one more point both context and redux can work simultaneously, you can keep context for any part of the DOM tree

**Comment by arp99** [comment]
Moreover the app currently lacks scoping in terms of CSS, it would be nicer we use CSS modules for each component, because keeping every class in the global namespace is not a good idea to scale

File: src/reducer/ThemeReducer.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export const ThemeReducer = (prevTheme, { type, payload }) => {
```
Comment (NegiAkash890) [inline src/reducer/ThemeReducer.js:1]: Shall we use default export here?
Since there will be only one ThemeReducer

Comment (arp99) [reply src/reducer/ThemeReducer.js:1]: Not much of a problem though, ok I will change it to default export

**Comment by NegiAkash890** [comment]
> Moreover the app currently lacks scoping in terms of CSS, it would be nicer we use CSS modules for each component, because keeping every class in the global namespace is not a good idea to scale

Can you resolve merge conflicts from your side?

**Comment by arp99** [comment]
> > Moreover the app currently lacks scoping in terms of CSS, it would be nicer we use CSS modules for each component, because keeping every class in the global namespace is not a good idea to scale
> 
> Can you resolve merge conflicts from your side?
More of such conflicts will happen in future when u merge PR s one by one

**Comment by arp99** [comment]
> > Moreover the app currently lacks scoping in terms of CSS, it would be nicer we use CSS modules for each component, because keeping every class in the global namespace is not a good idea to scale
> 
> Can you resolve merge conflicts from your side?

Do not push the package-lock.json it will create more and more merge conflicts which is not easy to resolve

**Comment by arp99** [comment]
> > > Moreover the app currently lacks scoping in terms of CSS, it would be nicer we use CSS modules for each component, because keeping every class in the global namespace is not a good idea to scale
> > 
> > 
> > Can you resolve merge conflicts from your side?
> 
> Do not push the package-lock.json it will create more and more merge conflicts which is not easy to resolve

For the code you pushed to main, it is hard to resolve conflicts as you have changed lot of component definitions you have to resolve the conflicts now.

**Comment by NegiAkash890** [comment]
Can u see screenshots of issue you are facing. Is it bcoz of linter? You can pull the main from the repo and install eslint. If you face any issue ping me?

**Comment by arp99** [comment]
> Can u see screenshots of issue you are facing. Is it bcoz of linter? You can pull the main from the repo and install eslint. If you face any issue ping me?

Yeah because of linter lots of files has changed, components with no body is directly returning the jsx wit brackets, making hard to edit for future, moreover the package-lock.json is huge to be edited, do not push package-lock.json

**Comment by arp99** [comment]
Refactor your code to use css modules that would not cause much problems

**Comment by arp99** [comment]
I am done with merging but there will be conflict with package-lock.json as It is hard to resolve

**Comment by NegiAkash890** [comment]
> I am done with merging but there will be conflict with package-lock.json as It is hard to resolve

Why package-lock.json is necessary?
Please read this [https://nodejs.dev/learn/the-package-lock-json-file](https://nodejs.dev/learn/the-package-lock-json-file)

**Comment by NegiAkash890** [comment]
> > Can u see screenshots of issue you are facing. Is it bcoz of linter? You can pull the main from the repo and install eslint. If you face any issue ping me?
> 
> Yeah because of linter lots of files has changed, components with no body is directly returning the jsx wit brackets, making hard to edit for future, moreover the package-lock.json is huge to be edited, do not push package-lock.json

Yup linters are difficult to setup and understand but it's necessary for the project at this time.

**Pull request merged**